sentry/src/test/java/io/sentry/SentryClientTest.kt

```diff
@@ -1592,6 +1594,234 @@ class SentryClientTest {
+        val sut = fixture.getSut()
+        val scope = createScope()
+
+        sut.captureTransaction(
+            SentryTransaction(fixture.sentryTracer),
+            scope,
+            Hints.withAttachment(fixture.attachment)
+        )
+
+        thenEnvelopeIsSentWith(0, 0, 1, 1)
+    }
+
+    @Test
+    fun `an attachment passed via hint is used with scope attachments for transactions`() {
+        val sut = fixture.getSut()
+
+        val scope = givenScopeWithStartedSession()
+        scope.addAttachment(fixture.attachment2)
+
+        sut.captureTransaction(
+            SentryTransaction(fixture.sentryTracer),
+            scope,
+            Hints.withAttachment(fixture.attachment)
+        )
+
+        thenEnvelopeIsSentWith(0, 0, 2, 1)
+    }
+
+    @Test
+    fun `can add to attachments in eventProcessor for transactions`() {
+        val sut = fixture.getSut { options ->
+            options.addEventProcessor(object : EventProcessor {
+                override fun process(event: SentryEvent, hints: Hints): SentryEvent? {
+                    return event
+                }
+
+                override fun process(
+                    transaction: SentryTransaction,
+                    hints: Hints
+                ): SentryTransaction? {
+                    assertEquals(listOf(fixture.attachment, fixture.attachment2), hints.attachmentContainer.all)
+                    hints.attachmentContainer.add(fixture.attachment3)
+                    return transaction
+                }
+            })
+        }
+
+        val scope = givenScopeWithStartedSession()
+        scope.addAttachment(fixture.attachment2)
+
+        sut.captureTransaction(
+            SentryTransaction(fixture.sentryTracer),
+            scope,
+            Hints.withAttachment(fixture.attachment)
+        )
+
+        thenEnvelopeIsSentWith(0, 0, 3, 1)
+    }
+
+    @Test
+    fun `can replace attachments in eventProcessor for transactions`() {
+        val sut = fixture.getSut { options ->
+            options.addEventProcessor(object : EventProcessor {
+                override fun process(event: SentryEvent, hints: Hints): SentryEvent? {
+                    return event
+                }
+
+                override fun process(
+                    transaction: SentryTransaction,
+                    hints: Hints
+                ): SentryTransaction? {
+                    hints.attachmentContainer.replaceAll(listOf(fixture.attachment))
+                    return transaction
+                }
+            })
+        }
+
+        val scope = givenScopeWithStartedSession()
+        scope.addAttachment(fixture.attachment2)
+
+        sut.captureTransaction(
+            SentryTransaction(fixture.sentryTracer),
+            scope,
+            Hints.withAttachment(fixture.attachment)
+        )
+
+        thenEnvelopeIsSentWith(0, 0, 1, 1)
+    }
+
+    @Test
+    fun `passing attachments via hint into breadcrumb ignores them`() {
```

Comment:
This test case and the one after it show that attachments in hints have no effect on breadcrumbs. I assume this is what we want. Correct?

Reply:
`h`: Why do we have a hint in `setBeforeBreadcrumb`? I don't get this. This might be frankly confusing. You could do `addAttachment` in `setBeforeBreadcrumb`, and nothing would happen.

Reply:
The hint for crumbs are only for filtering, they are not part of the event processing pipeline, I'd say that we don't even need this test.

Reply:
Is there anything we need to change here? Like passing a different Interface that doesn't have the attachment manipulation methods instead of `Hints`? Or is it ok if these are basically NOOP for breadcrumbs?